File: Score_After_Flipping_Matrix/Score_After_Flipping_Matrix.go
package main

import (
	"fmt"
	"math"
)

func main() {
	grid := [][]int{{0, 0, 1, 1}, {1, 0, 1, 0}, {1, 1, 0, 0}}
	fmt.Println(matrixScore(grid))
}

func matrixScore(grid [][]int) int {
	m := len(grid)
	n := len(grid[0])

	for i := 0; i < m; i++ {
		if grid[i][0] == 0 {
			for j := 0; j < n; j++ {
				grid[i][j] = 1 - grid[i][j]
			}
		}
	}

	for i := 0; i < n; i++ {
		count := 0
		for j := 0; j < m; j++ {
			if grid[j][i] == 1 {
				count += 1
			}
		}
		if count < m-count {
			for j := 0; j < m; j++ {
				grid[j][i] = 1 - grid[j][i]
			}
		}
	}
	res := 0.0
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			p := float64(n - j - 1)
			res += float64(grid[i][j]) * math.Pow(2, p)
		}
	}
	return int(res)
}

File: Score_After_Flipping_Matrix/Score_After_Flipping_Matrix.py
grid = [[0,0,1,1],[1,0,1,0],[1,1,0,0]]

def matrixScore(grid):
    m = len(grid)
    n = len(grid[0])
    for i in range(m):
        if grid[i][0] == 0:
            for j in range(n):
                grid[i][j] = 1 - grid[i][j]
    


    for i in range(n):
        count = 0
        for j in range(m):
            if grid[j][i] == 1:
                count+=1
        if count < m - count:
            for j in range(m):
                grid[j][i] = 1 - grid[j][i]
    
    res = 0
    for i in range(m):
        for j in range(n):
            res += grid[i][j] * 2 ** (n-j-1)
        
    return res
    

print(matrixScore(grid))